src/axom/primal/operators/detail/winding_number_2d_memoization.hpp

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2017-2025, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*!
+ * \file NURBSCurve.hpp
+ *
+ * \brief A NURBS curve primitive
+ */
+
+#ifndef AXOM_PRIMAL_WINDING_NUMBER_2D_MEMOIZATION_HPP_
+#define AXOM_PRIMAL_WINDING_NUMBER_2D_MEMOIZATION_HPP_
+
+#include "axom/core.hpp"
+#include "axom/slic.hpp"
+
+#include "axom/primal/geometry/KnotVector.hpp"
+#include "axom/primal/geometry/BezierCurve.hpp"
+#include "axom/primal/geometry/NURBSCurve.hpp"
+#include "axom/primal/geometry/Point.hpp"
+#include "axom/primal/geometry/Vector.hpp"
+#include "axom/primal/geometry/BoundingBox.hpp"
+
+#include "axom/primal/operators/is_convex.hpp"
+
+#include <vector>
+#include <ostream>
+#include <math.h>
+
+#include "axom/fmt.hpp"
+
+namespace axom
+{
+namespace primal
+{
+namespace detail
+{
+/*!
+ * \struct SubdivisionData
+ *
+ * \brief Stores BezierCurves and relevant data/flags
+ */
+template <typename T>
+struct SubdivisionData
+{
+  SubdivisionData() = default;
+
+  SubdivisionData(const BezierCurve<T, 2>& a_curve, bool knownConvex) : curve(a_curve)
+  {
+    isConvexControlPolygon = knownConvex ? true : is_convex(Polygon<T, 2>(curve.getControlPoints()));
+    boundingBox = curve.boundingBox();
+  }
+
+  BezierCurve<T, 2> curve;
+  bool isConvexControlPolygon;
+  BoundingBox<T, 2> boundingBox;
+};
```

Comment:
Naming: 
* Should the name include `Bezier`? 
* This struct doesn't have much to do w/ subdivision